src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/AddPackageReferenceCommandRunner.cs

```diff
@@ -204,24 +197,54 @@ private DependencyGraphSpec ReadProjectDependencyGraph(PackageReferenceArgs pack
             // If the user did not specify a version then write the exact resolved version
             if (packageReferenceArgs.NoVersion)
             {
-                // Get the flattened restore graph
-                var flattenedRestoreGraph = restorePreviewResult
-                    .Result
-                    .RestoreGraphs
-                    .First()
-                    .Flattened;
-
-                // Get the package entry and version from the graph
-                var packageEntry = flattenedRestoreGraph
-                    .Single(p => p.Key.Name.Equals(packageReferenceArgs.PackageDependency.Id, StringComparison.OrdinalIgnoreCase));
-                var resolvedVersion = packageEntry
-                    .Key
-                    .Version;
-
-                //Update the packagedependency with the new version
-                packageReferenceArgs.PackageDependency = new PackageDependency(packageReferenceArgs.PackageDependency.Id,
-                    VersionRange.Parse(resolvedVersion.ToString()));
+                // Get the package version from the graph
+                var resolvedVersion = privateGetPackageVersionFromRestoreResult(restorePreviewResult, packageReferenceArgs);
+
+                if (resolvedVersion != null)
+                {
+                    //Update the packagedependency with the new version
+                    packageReferenceArgs.PackageDependency = new PackageDependency(packageReferenceArgs.PackageDependency.Id,
+                        VersionRange.Parse(resolvedVersion.ToString()));
```

Comment:
There's no need to parse this as a string. Just pass the resolvedVersion as the min version.